src/util/get-github-client-config.ts

```diff
@@ -29,33 +31,34 @@ export async function getGitHubApiUrl(jiraHost: string, gitHubAppId: number) {
 const getGitHubClientConfigFromGitHubInstallationId = async (gitHubInstallationId: number): Promise<GitHubClientConfig> => {
 	const subscription = await Subscription.findOneForGitHubInstallationId(gitHubInstallationId);
 	const gitHubAppId = subscription?.gitHubAppId;
-	return getGitHubClientConfigFromAppId(gitHubAppId);
+	return getGitHubClientConfigFromAppId(gitHubAppId, subscription?.jiraHost);
 };
 
-const getGitHubClientConfigFromAppId = async (gitHubAppId: number | undefined): Promise<GitHubClientConfig> => {
+const getGitHubClientConfigFromAppId = async (gitHubAppId: number | undefined, jiraHost?: string): Promise<GitHubClientConfig> => {
 	const gitHubServerApp = gitHubAppId && await GitHubServerApp.getForGitHubServerAppId(gitHubAppId);
-	const gitHubCloudUrls = {
+	if (gitHubServerApp) {
+		return	{
+			hostname: gitHubServerApp.gitHubBaseUrl,
+			baseUrl: gitHubServerApp.gitHubBaseUrl,
+			appId: gitHubServerApp.appId,
+			privateKey: await gitHubServerApp.decrypt("privateKey")
+		};
+	}
+	// cloud config
+	const privateKey = await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost)? await keyLocator(): PrivateKey.findPrivateKey();
```

Comment:
Doesn't the `keyLocator` already have the logic to find the private key?  Why is this in a ternary operator?

Reply:
yes, `keyLocator` have the logic but want to put behind FF so that it can be tested throughly before rolling out.